### DISCUSSION TITLE
Is it possible to install PiVPN on CentOS?

### DISCUSSION THREAD

**Discussion by ghost** [pivpn/pivpn#1392] [Feature requests]
Hi,

I absolutely love PiVPN! I It was pleasantly surprising how it took only 2 minutes to set up PiVPN and its openvpn runs perfectly on all the clients I tested. In comparison, I gave up setup openvpn the old ways many times before, with one success which turned out had buggy implementation (Synology NAS' openvpn).

I understand pivpn can be quite compatible with debian-based systems, yet, another popular branch of OSes are CentOS / RHEL based. 

I tried to google how to set up pivpn on CentOS but could not find it, so can I ask if this is something possible? But if not, hey this is totally understandable, and I'm not even requesting it. Just curious.

Thank you!

**Comment by redfast00** [comment]
Currently, it cannot do an install on a distro that doesn't have apt as package manager. We are however working on a way to install on other distros.

**Comment by coolapso** [comment]
@redfast00 ; @0-kaladin ; I would like to start building the support for CentOS and perhaps Arch ARM later, However would like to ask you what you consider to be the best approach for this: 

1 - Unique install script by adding support for centOS (dnf / yum ) inside the existing script
2 - Make a separate script based on the existing one but for CentOS / Fedora / RHEL

**Comment by redfast00** [comment]
There has been a PR in the test branch which makes it easier to add support for other distros. I don't know how much centOS is different from Debian, but I think that as long as it's manageable, we should try to keep it in one script for ease of use. @0-kaladin: when will we merge the test branch into the master branch?

**Comment by 0-kaladin** [comment]
Def should stay one script. Pihole does it as well. I need to test more
before merging and there are comments in the PR that need addressing. The
past year I've been going through a lot with work and home life. Free time
for hobby projects has been nil. After I move, I'm hoping by Sept I'll
tackle a ton of changes and updates here.

**Comment by coolapso** [comment]
@redfast00 it's bit different, default firewall in centOS 7 is firewalld, when using iptables rules are in /etc/sysconfig/iptables , off course package manager which is yum (getting deprecated and being replaced with dnf). 
will take a look in test branch and pihole, and eventually start building based on test. 
:) ty

**Comment by coolapso** [comment]
@redfast00  @0-kaladin 

Since I am migrating my cloud servers to new servers in a new provider I will need to use OpenVPN in CentOS ... having this said I would love to have pivpn with me on this one ... so will start to work on this and eventually do a major code review and fix some issues around here!! =)

**Comment by redfast00** [comment]
Okay, glad to hear that. I'm currently taking exams, so I won't have a lot of time for code review. Good luck!

**Comment by coolapso** [comment]
~~redfast00 I dont have much time as well but will set aside stuff I have to do to do this one because I really want to use pivpn ... so ... will try to improve it and fix some stuff on the queue~~

**Comment by guest2** [comment]
The current pi-hole install script works great on CentOS. Can it be adapted to install PiVPN?

**Comment by coolapso** [comment]
@guest2 this is an improvement we are willing to do, but not taking priority at the moment, if anyone comes forward with it i don't mind to review, test and accept/request changes. 

unfortunately we are not many active maintainers and we don't either have a contributing community as big as pihole so things here have to go slower and small bits at a time  =/

**Comment by redfast00** [comment]
Sorry guys, just wanted to unsubscribe and misclicked

**Comment by computergeek125** [comment]
So, I found this project somewhat recently - but just about my whole homelab is running some version of EL over Debian/Ubuntu

On the subject of adding support for EL* in PiVPN, how much traction do you think there would be in the community for using it?  With the exception of Oracle which would need an exception on the `epel-release` package name, you can cover a lot of OSes that are recompiles of or highly adjacent to RHEL (Rocky, Oracle, RHEL, Fedora, CentOS, etc)

From a cursory glance through the script, it looks like whenever `apk` support was added that roughed out a lot of angles that would be needed to run on `dnf`, which seems to slide somewhere in between `apt` and `apk` in terms of which dependencies/commands it uses

**Reply by coolapso** [reply]
Not on the plans. 

Alpine support was added as it opens the opens the path for containers which should be the priority not supporting other Distros.
